engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetViewportSizeOf.java

```diff
@@ -86,6 +84,9 @@ public void update() {
 
     @Override
     public String toString() { // TODO: used for logging purposes at the moment, investigate different methods
+        if (isTheDefaultInstance()) { // necessary since default instance is generated before setFrameBuffersManager
+            fbo = frameBuffersManager.getFBO(fboName);
+        }
```

Comment:
I can see what you are doing here, attempting to avoid an NPE if fbo is still null. However, an instance of this class doesn't need knowledge of the width and height of the fbo except when it generates the task or updates it. 

So, I'd remove this IF block and I'd remove the printout of fbo.width() and fbo.height(). Those can stay in the task.

Also, I'd remove the private fbo variable: we can instantiate one briefly when we need to gets its width and height, in generateTask() and update().